File: LLM-Translations/GPT-4/avatar/Python/Java/atcoder_ABC158_B.java

import java.util.Scanner;

public class atcoder_ABC158_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N, A, B, mod = 1000000007, ans = 0, rep, res;
        N = sc.nextInt();
        A = sc.nextInt();
        B = sc.nextInt();
        sc.close();
        
        rep = N / (A + B);
        ans += rep * A;
        res = N - rep * (A + B);
        ans += Math.min(res, A);

        System.out.println(ans);
    }
}

//End of Code

File: LLM-Translations/GPT-4/avatar/Python/Go/atcoder_ABC158_B.go
package main

import (
	"fmt"
)

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	var N, A, B int
	fmt.Scanf("%d %d %d\n", &N, &A, &B)
	
	mod := 10^9 + 7
	ans := 0
	rep := N / (A + B)
	ans += rep * A
	res := N - rep*(A + B)
	ans += min(res, A)
	fmt.Println(ans)
}

// End of Code.
